Add readme for sql server (#30466)
dotnet/efcore: src/EFCore.SqlServer/README.md

> Microsoft.EntityFrameworkCore.SqlServer is Microsoft SQL Server EF Core Database Provider, which enables you to use Entity Framework Core with Microsoft SQL Server (including Azure SQL Database).

Reviewer: ```suggestion
Microsoft.EntityFrameworkCore.SqlServer is the database provider for Microsoft SQL Server and Azure SQL. This providers allows you to use Entity Framework Core with Microsoft SQL Server and Azure SQL databases.
```

## Getting started

> Explain how to use your package, provide clear and concise getting started instructions, including any necessary steps.

Reviewer: This seems like placeholder text.

### Prerequisites

- Supported database Engines: Microsoft SQL Server (2012 onwards)

> - The provider references Microsoft.Data.SqlClient (not System.Data.SqlClient). If your project takes a direct dependency on SqlClient, make sure it references the Microsoft.Data.SqlClient package.

Reviewer: Make these links?

## Usage

Once you've installed the package, you can use it in your Entity Framework Core application by specifying the SQL Server provider in your DbContext's OnConfiguring method:

```csharp
protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
{
    optionsBuilder.UseSqlServer("Server=(localdb)\\mssqllocaldb;Database=MyDatabase;Trusted_Connection=True;");
}
```

> In this example, we're using the (localdb)\mssqllocaldb server with the MyDatabase database. You'll need to adjust the connection string to match your own SQL Server instance and database.

Reviewer: Link to guidance on this?

## Features

> The SQL Server provider for Entity Framework Core supports a variety of features, including:
> 
> - Basic CRUD operations
> - Transactions
> - Stored procedures
> - Views
> - Table-valued functions
> - Identity columns
> - In-memory tables

Reviewer: ```suggestion
The SQL Server provider  supports all common features of [Entity Framework Core](https://learn.microsoft.com/ef/core/) as well as some [SQL Server-specific features](https://learn.microsoft.com/ef/core/providers/sql-server/?tabs=dotnet-core-cli) including temporal tables and memory-optimized tables.
```
